foundations/src/security/mod.rs

```diff
@@ -348,6 +370,27 @@ pub enum Rule {
 pub fn enable_syscall_sandboxing(
     violation_action: ViolationAction,
     exception_rules: &Vec<Rule>,
+) -> BootstrapResult<()> {
+    let current_mode = get_current_thread_seccomp_mode()?;
+    if current_mode != SeccompMode::None {
+        return Ok(());
```

Comment:
Silently succeeding seems like the wrong method here, if the exception list is different to what's actually applied you could cause unexpected failures later on. I would feel happier with an error that we can detect and report in a more sensible manner (and/or the app can detect this is needed)

Reply:
Yeah, I debated that myself. It now returns an Error. Since we need the caller to be able to distinguish this from other things going wrong during initialization, I introduced an error type for this. It should make for nice discoverability when using this API for the first time, and still gives callers the choice of what to do (whereas previously seccomp most likely would have just terminated the process).